internal/terraform/node_resource_plan_instance.go

```diff
@@ -157,12 +157,32 @@ func (n *NodePlannableResourceInstance) managedResourceExecute(ctx EvalContext)
 
 	importing := n.importTarget.ID != ""
 
+	if importing && n.Config == nil && !n.generateConfig {
+		// Then the user wrote an import target to a target that didn't exist.
+		if n.Addr.Module.IsRoot() {
+			diags = diags.Append(&hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Import block target does not exist",
+				Detail:   "The target for the given import block does not exist. If you wish to automatically generate config for this resource, use the -generate-config-out option within terraform plan. Otherwise, make sure the target resource exists within your configuration.",
```

Comment:
Could we color code this so its highlighted more effectively? Perhaps even at the bottom as the last text output? I believe this will also be written alongside the error message of "resource block is not present" in red.

Reply:
Diagnostics have very rigid formatting because we have multiple different renderers for them, such as the Terraform Cloud web UI.

We could potentially make this be something other than a diagnostic but it would require defining a special representation of this in the structured log output so that Terraform Cloud's web UI, and anything else using the structured run output, can recognize it and render it.

Using a normal diagnostic avoids the need to change each UI (including ones we don't control) separately.

Reply:
Got it, nvm then

Reply:
<img width="1084" alt="Screenshot 2023-05-12 at 08 21 44" src="https://github.com/hashicorp/terraform/assets/56838463/9ebaefc5-4381-4e8d-a5f5-efb8c5f2222b">

Reply:
That's an example of how this would render.

Reply:
It would be nice to have an example of the new plan command as the last line of the diagnostic, perhaps separated by a new line, for easy copying.

```
Otherwise, make sure the target resource exists within your configuration. For example:

  terraform plan -generate-config-out=generated.tf
```

Reply:
Done!

Reply:
It's a bit awkward for a diagnostic generated inside Terraform Core (rather than Terraform CLI) to be emitting a command to run, because at this layer we can't rely on the "running in automation" flag to suppress the suggested command for folks who are running Terraform indirectly through other software, like Terraform Cloud. :confounded: 

But I'm also not convinced it's worth the additional complexity to try to invert control here so that it's the CLI layer intercepting this message and rewording it. I suppose the pragmatic thing to do here is to leave this as currently implemented and then see (based on feedback) whether it's worth doing something more complex here.

---

In case we _do_ decide to try to improve on this in future and someone ends up looking here to see what I was talking about :grin: the "running in automation" concept I'm talking about is represented by this field in the `command` package, which we also propagate into a few other places that we consider to be parts of "Terraform CLI" as opposed to "Terraform Core":

https://github.com/hashicorp/terraform/blob/2a49d4b7844d464404b27b43e23c3ee3cb6eef74/internal/command/meta.go#L84-L95

This flag is controlled externally by an environment variable we've documented as something folks should set when they run Terraform indirectly through other automation, in [Controlling Terraform Output in Automation](https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform#controlling-terraform-output-in-automation).